config/runtimes/kserve-huggingfaceserver-multinode.yaml

```diff
@@ -0,0 +1,210 @@
+              fi & # Run this check in the background
+              p3=$!
+
+              # Wait for all processes to complete
+              wait $p1 $p2 $p3
+              wait $p1 && wait $p2 && wait $p3
+      startupProbe:
+        failureThreshold: 40
+        periodSeconds: 30
+        successThreshold: 1
+        timeoutSeconds: 5
+        exec:
+          command:
+            - bash
+            - -c
+            - |
+              ray status
+  volumes:
+    - name: shm
+      emptyDir:
+        medium: Memory
+        sizeLimit: 12Gi
+  workerSpec:
+    containers:
+      - name: worker-container
+        image: kserve/huggingfaceserver:latest
+        command: ["bash", "-c"]
+        args:
+          - |
+            SECONDS=0
+
+            while true; do              
+              if (( SECONDS <= 120 )); then
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379" > /dev/null 2>&1; then
+                  echo "GCS is ready."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Waiting for GCS to be ready."
+              else
+                if ray health-check --address "${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                  echo "GCS is ready. Any error messages above can be safely ignored."
+                  break
+                fi
+                echo "$SECONDS seconds elapsed: Still waiting for GCS to be ready."
+                echo "For troubleshooting, refer to the FAQ at https://github.com/ray-project/kuberay/blob/master/docs/guidance/FAQ.md."
+              fi
+              
+              sleep 5
+            done
+
+            echo "Attempting to connect to Ray cluster at $RAY_HEAD_ADDRESS ..."
+            RAY_HEAD_ADDRESS="${ISVC_NAME}-head.${POD_NAMESPACE}.svc.cluster.local:6379"
+            ray start --address="$RAY_HEAD_ADDRESS" --block
+        env:
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace              
+        resources:
+          limits:
+            cpu: "16"
+            memory: 48Gi
+          requests:
+            cpu: "8"
+        volumeMounts:
+          - name: shm
+            mountPath: /dev/shm
+        livenessProbe:
+          failureThreshold: 3
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 5
+          exec:
+            command:
+              - bash
+              - -c
+              - |
+                [[ $(ray status | grep -c node_) -eq ${PIPELINE_PARALLEL_SIZE} ]]
+        startupProbe:
+          failureThreshold: 12
+          periodSeconds: 5
+          successThreshold: 1
+          timeoutSeconds: 5
+          exec:
+            command:
+              - ray
+              - status
```

Comment:
what does it return? status code or messages?

Reply:
It returns ray status information but what this is looking for is return.code 0.